core/store/src/trie/split.rs

```diff
@@ -334,50 +369,57 @@ where
         // of nibbles is required. If we ended with an odd number of nibbles, we can try descending
         // into the first child.
         if self.nibbles.len() % 2 != 0 {
-            self.force_next_step();
-            if let Some(current_split) = self.best_split_at_current_path() {
+            self.force_next_step()?;
+            if let Some(current_split) = self.best_split_at_current_path()? {
                 if current_split.mem_diff() < best_split.mem_diff() {
                     best_split = current_split;
                 }
             }
         }
 
-        debug_assert!(best_split.mem_diff() < u64::MAX);
-        best_split
+        if best_split.mem_diff() < u64::MAX {
```

Comment:
Can you instead compare to dummy? Or add a method is_valid? The dummy abstraction leaked here in the for of u64 max